notification/app/notification/services/fcm/AndroidConfigConverter.scala

```diff
@@ -0,0 +1,150 @@
+        Keys.UriType -> patformLink.`type`,
+        Keys.Uri -> patformLink.uri
+      ) ++ Map(
+        Keys.Section -> sectionLink.map(new URI(_)).map(_.toString),
+        Keys.Edition -> edition.map(_.toString),
+        Keys.Keyword -> keyword.map(_.toString),
+        Keys.ImageUrl -> breakingNews.imageUrl.map(_.toString),
+        Keys.ThumbnailUrl -> breakingNews.thumbnailUrl.map(_.toString)
+      ).flattenValues
+    )
+  }
+
+  private def toContent(cn: ContentNotification): FirebaseAndroidNotification = {
+    val link = toPlatformLink(cn.link)
+
+    FirebaseAndroidNotification(
+      Map(
+        Keys.Type -> AndroidMessageTypes.Custom,
+        Keys.UniqueIdentifier -> cn.id.toString,
+        Keys.Title -> cn.title,
+        Keys.Ticker -> cn.message,
+        Keys.Message -> cn.message,
+        Keys.Link -> toAndroidLink(cn.link).toString,
+        Keys.Topics -> cn.topic.map(toAndroidTopic).mkString(","),
+        Keys.UriType -> link.`type`,
+        Keys.Uri -> new URI(link.uri).toString,
+        Keys.Debug -> conf.debug.toString
+      ) ++ Map(
+        Keys.ThumbnailUrl -> cn.thumbnailUrl.map(_.toString)
+      ).flattenValues
+    )
+  }
+
+  private def toMatchStatusAlert(matchStatusAlert: FootballMatchStatusNotification): FirebaseAndroidNotification = FirebaseAndroidNotification(
+    Map(
+      "type" -> Some(AndroidMessageTypes.FootballMatchAlert),
+      "homeTeamName" -> Some(matchStatusAlert.homeTeamName),
+      "homeTeamId" -> Some(matchStatusAlert.homeTeamId),
+      "homeTeamScore" -> Some(matchStatusAlert.homeTeamScore.toString),
+      "homeTeamText" -> Some(matchStatusAlert.homeTeamMessage),
+      "awayTeamName" -> Some(matchStatusAlert.awayTeamName),
+      "awayTeamId" -> Some(matchStatusAlert.awayTeamId),
+      "awayTeamScore" -> Some(matchStatusAlert.awayTeamScore.toString),
+      "awayTeamText" -> Some(matchStatusAlert.awayTeamMessage),
+      "currentMinute" -> Some(""),
+      "important" -> Some(matchStatusAlert.importance.toString),
+      "matchStatus" -> Some(matchStatusAlert.matchStatus),
+      "matchId" -> Some(matchStatusAlert.matchId),
+      "matchInfoUri" -> Some(matchStatusAlert.matchInfoUri.toString),
+      "articleUri" -> matchStatusAlert.articleUri.map(_.toString),
+      "competitionName" -> matchStatusAlert.competitionName,
+      "venue" -> matchStatusAlert.venue
+    ).flattenValues
+  )
+
+  case class PlatformUri(uri: String, `type`: String)
+
+  protected def toPlatformLink(link: Link): PlatformUri = link match {
+    case Link.Internal(contentApiId, _, _) => PlatformUri(s"x-gu:///items/$contentApiId", "item")
+    case Link.External(url) => PlatformUri(url, "external")
+  }
+
+  private def toAndroidLink(link: Link) = link match {
+    case Link.Internal(contentApiId, _, _) => new URI(s"x-gu://www.guardian.co.uk/$contentApiId")
+    case Link.External(url) => new URI(url)
+  }
+
+  private[services] def toTags(destination: Destination) = destination match {
```

Comment:
I think this can just be
`private[services] def toTags(destination: Destination) = Some(destination.fold(Tags.fromTopics(_), Tags.fromUserId(_)))`

Reply:
I quite like to keep a pattern match, I find it more readable